### PR TITLE
add EL9 x86_64 (based on alma)

### PR DESCRIPTION
**Pull request by halfcyan**

### CODE REVIEW

File: epel9.tpl [hunk 1]
```diff
@@ -0,0 +1,60 @@
+config_opts['chroot_setup_cmd'] += " epel-rpm-macros"
```
Comment (madonuko) [inline epel9.tpl:1]: here?

Comment (korewaChino) [reply epel9.tpl:1]: this would literally just write `dnf epel-rpm-macros`

**Review by korewaChino — CHANGES_REQUESTED** [review]
change the chroot_setup_cmd so it installs properly

File: epel9.tpl (hunk 1 above)
Comment (halfcyan) [reply epel9.tpl]: I think what's going on here is that the template is trying to add that package to the install command on the EL9 template. I removed that line in the EPEL template and moved the `epel-rpm-macros` package to be installed with the chroot setup cmd on the EL9 template. At this point, I'm just unsure whether I need to keep the `config_opts['chroot_setup_cmd'] = 'install @{% if mirrored %}buildsys-{% endif %}build'` line.

**Comment by halfcyan** [comment]
It might be worth it to combine the EL9 and EPEL templates. That would simplify it so EL only has one template. Additionally, it could remove errors (such as the earlier chroot setup command issue). I don't think we ever plan on doing this without EPEL.

**Comment by lleyton** [comment]
I agree, there's no need to have EPEL as seperate

File: terra-el9-aarch64.cfg [hunk 2]
```diff
@@ -0,0 +1,6 @@
+config_opts['releasever'] = 'el9'
```
Comment (lleyton) [inline terra-el9-aarch64.cfg:1]: This one needs to be just `9` as well

File: terra-el9-aarch64.cfg [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+config_opts['target_arch'] = 'aarch64'
+config_opts['legal_host_arches'] = ('aarch64',)
+
+include('templates/el9.tpl')
+include('templates/epel9.tpl')
```
Comment (lleyton) [inline terra-el9-aarch64.cfg:6]: remove epel

File: el9.tpl [hunk 4]
```diff
@@ -0,0 +1,214 @@
+config_opts['root'] = 'terra-{{ releasever }}-{{ target_arch }}'
```
Comment (lleyton) [inline el9.tpl:1]:
```suggestion
config_opts['root'] = 'terra-el{{ releasever }}-{{ target_arch }}'
```
that way we can distinguish it from Fedora Terra

File: el9.tpl [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+config_opts['package_manager'] = 'dnf'
+config_opts['extra_chroot_dirs'] = [ '/run/lock', ]
+config_opts['chroot_setup_cmd'] = 'install @{% if mirrored %}buildsys-{% endif %}build'
+config_opts['plugin_conf']['root_cache_enable'] = True
+config_opts['plugin_conf']['yum_cache_enable'] = True
+config_opts['plugin_conf']['ccache_enable'] = True
+config_opts['plugin_conf']['ccache_opts']['compress'] = 'on'
+config_opts['plugin_conf']['ccache_opts']['max_cache_size'] = '10G'
+config_opts['chroot_setup_cmd'] = 'install bash bzip2 coreutils cpio diffutils redhat-release findutils gawk glibc-minimal-langpack grep gzip info patch redhat-rpm-config rpm-build sed tar unzip util-linux which xz epel-rpm-macros'
+config_opts['dist'] = 'el9'  # only useful for --resultdir variable subst
```
Comment (lleyton) [inline el9.tpl:11]:
```suggestion
config_opts['dist'] = 'el{{ releasever }}'  # only useful for --resultdir variable subst
```

File: el9.tpl [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+config_opts['releasever'] = '9'
```
Comment (lleyton) [inline el9.tpl:12]: don't define releasever here, those are defined in the individual .cfg files

File: el9.tpl [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+config_opts['bootstrap_image'] = 'quay.io/almalinuxorg/almalinux:9'
```
Comment (lleyton) [inline el9.tpl:13]:
```suggestion
config_opts['bootstrap_image'] = 'quay.io/almalinuxorg/almalinux:{{ releasever }}'
```

**Review by lleyton — CHANGES_REQUESTED** [review]

File: el9.tpl (hunk 4 above)
Comment (halfcyan) [reply el9.tpl:1]: removed el there originally because i thought the releasever needed it, forgot to readd that

File: terra-el9-aarch64.cfg [hunk 8]
```diff
@@ -0,0 +1,5 @@
+config_opts['releasever'] = 'e'
```
Comment (lleyton) [inline terra-el9-aarch64.cfg:1]:
```suggestion
config_opts['releasever'] = '9'
```

typo?

File: el9.tpl [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+gpgcheck=0
+assumeyes=1
+syslog_ident=mock
+syslog_device=
+metadata_expire=0
+mdpolicy=group:primary
+best=1
+install_weak_deps=0
+protected_packages=
+module_platform_id=platform:el9
```
Comment (lleyton) [inline el9.tpl:32]:
```suggestion
module_platform_id=platform:el{{ releasever }}
```

File: el9.tpl [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+user_agent={{ user_agent }}
+
+
+[baseos]
+name=AlmaLinux $releasever - BaseOS
+mirrorlist=https://mirrors.almalinux.org/mirrorlist/$releasever/baseos
+# baseurl=https://repo.almalinux.org/almalinux/$releasever/BaseOS/$basearch/os/
+enabled=1
+gpgcheck=1
+gpgkey=file:///usr/share/distribution-gpg-keys/alma/RPM-GPG-KEY-AlmaLinux-9
```
Comment (lleyton) [inline el9.tpl:42]:
```suggestion
gpgkey=file:///usr/share/distribution-gpg-keys/alma/RPM-GPG-KEY-AlmaLinux-{{ releasever }}
```

Comment (lleyton) [reply el9.tpl:42]: you'll want to do these for the other GPG keys as well

**Review by lleyton — CHANGES_REQUESTED** [review]
can you rename the el9.tpl config to just terra-el.tpl (and update the paths)

the template should be agnostic to the enterprise linux version being used

**Pull request merged**